src/gleam_stdlib.erl

```diff
@@ -319,3 +320,58 @@ print(String) ->
 println(String) ->
     io:put_chars([String, $\n]),
     nil.
+
+inspect(true) ->
+    "True";
+inspect(false) ->
+    "False";
+inspect(Any) when is_atom(Any) ->
+    lists:map(
+        fun(Part) ->
+            [Head | Tail] = string:next_grapheme(unicode:characters_to_binary(Part)),
+            [string:uppercase([Head]), Tail]
+        end,
+        re:split(erlang:atom_to_list(Any), "_+", [{return, iodata}])
+    );
+inspect(Any) when is_integer(Any) ->
+    erlang:integer_to_list(Any);
+inspect(Any) when is_float(Any) ->
+    io_lib_format:fwrite_g(Any);
+inspect(Any) when is_binary(Any) ->
+    Pattern = [$"],
+    Replacement = [$\\, $\\, $"],
+    Escaped = re:replace(Any, Pattern, Replacement, [{return, binary}, global]),
+    ["\"", Escaped, "\""];
+inspect(Any) when is_list(Any) ->
+    ["[",
+        lists:join(<<", ">>,
+            lists:map(fun inspect/1, Any)
+        ),
+    "]"];
+inspect(Any) when is_tuple(Any) % Type constructors
+  andalso is_atom(element(1, Any))
+  andalso element(1, Any) =/= false
+  andalso element(1, Any) =/= true
+  andalso element(1, Any) =/= nil
+->
+    [Atom | ArgsList] = erlang:tuple_to_list(Any),
+    Args =
+        lists:join(<<", ">>,
+            lists:map(fun inspect/1, ArgsList)
+    ),
+    [(inspect(Atom)), "(", Args, ")"];
+inspect(Any) when is_tuple(Any) ->
+    ["#(",
+        lists:join(<<", ">>,
+            lists:map(fun inspect/1, erlang:tuple_to_list(Any))
+        ),
+    ")"];
+inspect(Any) when is_function(Any) ->
+    {arity, Arity} = erlang:fun_info(Any, arity),
+    ArgsAsciiCodes = lists:seq(97, 97 + Arity - 1), % a = ASCII 97
+    Args = lists:join(<<", ">>,
+        lists:map(fun(Arg) -> <<Arg>> end, ArgsAsciiCodes)
+    ),
+    ["//fn(", Args, ") { ... }"];
+inspect(_Any) ->
+    ["//erl("] ++ erlang:pid_to_list(erlang:self()) ++ [")"].
```

Comment:
I don't grasp the purpose of this, so I don't know where you are heading.

```erlang
["//erl(", io_lib:format("~p", ["AnyValueHere"]), ")"].
```

... would show:

```erlang
["//erl(",["\"AnyValueHere\""],")"]
```

Is this what you want? I thought you wanted some pid style syntax? How would I get a process id of a variable (I wouldn't know that that's even possible)?

I will change it to this and hopefully it will be what you wanted in the first place.

Reply:
Yup, that's right. I want the thing in Erlang syntax, no matter what it is. This matches what the JS version does.

I used the pid as an example, I didn't want to convert things to pids! Sorry is that was unclear.

Reply:
If this is correct now all is forgiven :phew:, I am more worried about wasting your time with all the back and forth.